### PR TITLE
Add metaclass for custom tensor

### PR DESCRIPTION
**Pull request by gmuraru**
## Description
Create a metaclass that would add the methods and properties to the custom tensor.
This is needed when creating the AST and doing remote operations (TODO: need to add an example).

## Affected Dependencies
- Custom tensors

## How has this been tested?
- Added test to validate properties
- Added test to validate that only the forwarded methods work

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests
- [ ] Add remote tensor tests

### CODE REVIEW

File: src/syft/util.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ def get_subclasses(obj_type: type) -> List[type]:
 
     """
 
-    classes = list()
+    classes = []
```
Comment (gmuraru) [inline src/syft/util.py:61]: They have nothing to do with the PR description.
Those were some simple fixes shown by flake

**Comment by koenvanderveen** [comment]
Looks good! it would be good to have tests that check if this now actually works remotely, but I suppose we need `.send()` to be working first.

**Review by koenvanderveen — APPROVED** [review]

**Comment by madhavajay** [comment]
@gmuraru This looks great, whats left to do? Serialization? Or should we break into a separate task?

**Comment by gmuraru** [comment]
Would close this for now and when more people would need this custom tensor there should be adapter from here: https://github.com/OpenMined/SyMPC/pull/89

**Pull request closed without merging**